### PR TITLE
[PoC] remove loading and new new data state and add loading flag to StoreResponse.Data

### PR DESCRIPTION
**Pull request by eyalgu**

### CODE REVIEW

File: store/src/main/java/com/dropbox/android/external/store4/impl/FetcherController.kt [hunk 1]
```diff
@@ -83,8 +83,6 @@ internal class FetcherController<Key : Any, Input : Any, Output : Any>(
                             origin = ResponseOrigin.Fetcher
                         )
                     }
-                }.onEmpty {
-                    emit(StoreResponse.NoNewData(ResponseOrigin.Fetcher))
```
Comment (eyalgu) [inline store/src/main/java/com/dropbox/android/external/store4/impl/FetcherController.kt:87]: this still needs to be addressed - if the fetcher is empty we will never revert from loading state to non loading state

**Comment by yigit** [comment]
i feel like we are turning in circles here as this makes it no longer stream of events but state. I would much prefer to solve it as an overloaded helper instead.

**Comment by yigit** [comment]
note that this requirement is also necessary for StoreResponse.error as you want to know if there is data even if error happened.

**Comment by NahroTo** [comment]
@yigit 
> i feel like we are turning in circles here as this makes it no longer stream of events but state

But isn't getting the state of the data more preferable than getting events? Especially since the readme states that Store adheres to a unidirectional data flow (state down, events up).

**Pull request closed without merging**